### PR TITLE
Multi-platform Dockerfile

### PR DESCRIPTION
**Pull request by kroese**
## Description

The backend supports ARM now, but that is not very useful as long as the frontend does not.

So this PR lets the frontend Dockerfile be build for ARM64 (and also x86 CPU's missing the SSE4 instruction set).

It is based on the work of [ubergeek77](https://github.com/ubergeek77/), all credits to him.

Fixes: https://github.com/LemmyNet/lemmy-ui/issues/1566
Fixes: https://github.com/LemmyNet/lemmy-ui/issues/2106

It also adds the propers labels, adds a healthcheck to monitor the container, and switches Inferno to production-mode which should increase performance.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,16 +1,41 @@
+FROM golang as go-builder
```
Comment (SleeplessOne1917) [inline Dockerfile:1]: What is golang needed for?

File: Dockerfile [hunk 2]
```diff
@@ -1,16 +1,41 @@
...
 
-ENV npm_config_target_arch=x64
 ENV npm_config_target_platform=linux
 ENV npm_config_target_libc=musl
 
 # Cache deps
 COPY package.json yarn.lock ./
-RUN yarn --production --prefer-offline --pure-lockfile
+
+RUN sed -i 's|"sharp": "^0.32.4"|"sharp": "^0.32.6"|g' package.json
```
Comment (SleeplessOne1917) [inline Dockerfile:36]: Should the repo just update to a newer version of `sharp`?

Comment (SleeplessOne1917) [reply Dockerfile:36]: Made a PR for the dependency upgrade.

**Comment by dessalines** [comment]
Actually I can't because for some reason edits by maintainers aren't being allowed.

**Comment by dessalines** [comment]
Yeah, I think it might just be a checkmark somewhere on the pull request specifically. https://stackoverflow.com/questions/63341296/github-pull-request-allow-edits-by-maintainers#63341866

**Comment by dessalines** [comment]
Question, node-prune the only thing that requires go correct? Because if that's the case I'm in favor of just removing it, since development is dead on it. I'm sure there are better ways to minimize node docker images now.

**Comment by dessalines** [comment]
I fixed some issues. Another build started.

**Review by dessalines — APPROVED** [review]

**Comment by dessalines** [comment]
I increased the timeout from 1h to 2h and restarted it. Not sure why these ones take so long, not even the CI for the lemmy back-end docker builds are this slow.

**Comment by dessalines** [comment]
It passed : https://hub.docker.com/layers/dessalines/lemmy-ui/0.19.0-alpha.2/images/sha256-200e55a928dce5b9460f674d584b1124a761d2e700beed8c11d20b7495af76ee?context=explore

**Pull request merged**